saas/management/commands/report_weekly_revenue.py

```diff
@@ -49,172 +49,236 @@ def add_arguments(self, parser):
             dest='at_time', default=None,
             help='Specifies the time at which the command runs'
         )
-        parser.add_argument('--provider', action='append',
+        parser.add_argument(
+            '--provider', action='append',
             dest='providers', default=None,
-            help='Specifies provider to generate reports for.')
+            help='Specifies provider to generate reports for.'
+        )
+        parser.add_argument(
+            '--period', action='store',
+            dest='period', default='weekly',
+            choices=['hourly', 'daily', 'weekly', 'monthly', 'yearly'],
```

Comment:
It might be best to construct the choices from `Plan.INTERVAL_CHOICES` like is done in serializers `EnumField`.

See:
https://github.com/djaodjin/djaodjin-saas/blob/3a2752cb0787ddf0562fc8ec7f091c7731174044/saas/api/serializers.py#L632C19-L634C54

and 
https://github.com/djaodjin/djaodjin-saas/blob/3a2752cb0787ddf0562fc8ec7f091c7731174044/saas/models.py#L2676